### PR TITLE
feat(player):  player 组件

### PR DESCRIPTION
**Pull request by tangying1027**
keyframes

### CODE REVIEW

File: packages/f-engine/test/timeline/keyFrames.test.tsx [hunk 1]
```diff
@@ -0,0 +1,267 @@
...
+    const context = createContext('子组件');
+    const { props } = (
+      <Canvas context={context}>
+        <Player
+          state="play"
+          keyFrames={[
+            {
+              view: {
+                to: '40px',
+                key: 'width',
```
Comment (zengyue) [inline packages/f-engine/test/timeline/keyFrames.test.tsx:112]: to: { width: '40px' , opacity: 0.5, height: xx}

File: packages/f-engine/src/playerFrames.ts [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+    }
+    // TODO
+    return {
+      ...child,
+      props: {
+        ...child?.props,
+        ...transStyle,
+        children: handerFrames(cur, child?.props?.children),
+      },
+    };
```
Comment (zengyue) [inline packages/f-engine/src/playerFrames.ts:24]:
```
function generateFrameElement(element, cur: Record<string, playerFrame>) {
  return Children.map(element, (child) => {
     if (!child) return child;
    const { key, props } = child;
    const newProps = cur[key] ? cur[key].to : {};
    const children = generateFrameElement(props.children);
    return Children.cloneElement(child, { ...newProps, children });
 };
```

File: packages/f-engine/test/timeline/keyFrames.test.tsx (hunk 1 above)
Comment (tangying1027) [reply packages/f-engine/test/timeline/keyFrames.test.tsx]: 已修改

File: packages/f-engine/src/playerFrames.ts (hunk 2 above)
Comment (tangying1027) [reply packages/f-engine/src/playerFrames.ts]: 已修改

File: packages/f-engine/src/player.ts [hunk 3]
```diff
@@ -19,34 +16,93 @@ export interface PlayerProps {
...
 class Player extends Component<PlayerProps> {
+  playerFrames;
+  index: number;
+  onend: Function;
+  constructor(props) {
+    super(props);
+    const { keyFrames = [], children } = props;
+    this.playerFrames = [];
+
+    keyFrames.forEach((cur) => {
```
Comment (zengyue) [inline packages/f-engine/src/player.ts:36]: reduce

**Pull request merged**